### PR TITLE
add gen fragments at 13.6 TeV for BPH-TRK

### PR DESCRIPTION
**Pull request by mtosi**
#### PR description:
add new genFragments at 13.6 TeV for BPH and TRK
- B0ToJpsiK0s w/ Jspi to muons
- DStarToD0Pi w/ D0 To KsPiPi
- LbToJpsiLambda w/ Jspi to muons
- LbToJpsiXiK0sPi w/ Jspi to muons
- OmegaMinus
- XiMinus

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
urgent 
   * this is need from TRK POG and BPH PAG in CMSSW_12_5_0 to perform V0 tracking optimization studies

**Comment by mmusich** [comment]
@cms-sw/generators-l2 it would be important that this update enters 12_5_0, because it will allow to ask samples for further tracking development targeting V0 reconstruction. Can you please review it?
Thanks much in advance
Att: @cms-sw/orp-l2

**Comment by perrotta** [comment]
> @cms-sw/generators-l2 it would be important that this update enters 12_5_0, because it will allow to ask samples for further tracking development targeting V0 reconstruction. Can you please review it? Thanks much in advance Att: @cms-sw/orp-l2

@cms-sw/generators-l2 please notice that we did not build pre5 yet only because we have issues with fwlite see https://github.com/cms-sw/cmssw/issues/39212
That means that you could still have some time, but not that much: we will not wait for this PR, and as soon as that issue is solved we'll start building pre5

**Comment by mtosi** [comment]
@cms-sw/generators-l2 are there issues w/ this PR ?
thanks

**Comment by perrotta** [comment]
@SiewYan, @mkirsano, @Saptaparna, @alberto-sanchez, @menglu21, @GurpreetSinghChahal : is anybody around for @cms-sw/generators-l2 able to review and sign this and the other pending PRs ?

**Comment by SiewYan** [comment]
Hello @perrotta , I am ok with the technical implementation, we need L2 @Saptaparna , @menglu21 , @GurpreetSinghChahal to sign off if the naming of the fragment is coherent with the practice.

File: Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Timing = 0.015818 sec/event
+# Event size = 456.1 kB/event
+
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.Pythia8CUEP8M1Settings_cfi import *
+
+
+generator = cms.EDFilter("Pythia8GeneratorFilter",
```
Comment (srimanob) [inline Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: Following the guideline (*), the generator filter does not include ExternalDecay, should Pythia8ConcurrentGeneratorFilter be used instead?

(*) https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookGenMultithread#Pythia8ConcurrentGeneratorFilter

File: Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Timing = 0.298047 sec/event
+# Event size = 410.3 kB/event
+
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.Pythia8CUEP8M1Settings_cfi import *
+
+
+generator = cms.EDFilter("Pythia8GeneratorFilter",
```
Comment (srimanob) [inline Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: Same question as OmegaMinus.

File: Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py (hunk 1 above)
Comment (srimanob) [reply Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: By the way, why TuneCUEP8M1? Not CP5?

File: Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py (hunk 2 above)
Comment (Saptaparna) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: Also, please use `Pythia8ConcurrentGeneratorFilter` to enable multithreading.

Comment (Saptaparna) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: Actually, perhaps with EvtGen this is not possible.

Comment (srimanob) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py:12]: This one should be used with `Pythia8ConcurrentGeneratorFilter`

**Comment by mtosi** [comment]
as suggested (thanks, I completely missed this option), 
I changed the 2 genFragments which do not depend on ExternalDecays parameters to use the multithreading `Pythia8ConcurrentGeneratorFilter`
while I did not patch the others, which make use of EvtGen

File: Configuration/Generator/python/#OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py# [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Found 37 output events for 40000 input events.
+# Filter efficiency = 0.000925
+# Timing = 0.015818 sec/event
+# Event size = 456.1 kB/event
```
Comment (mmusich) [inline Configuration/Generator/python/#OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py#:4]: This file shouldn't be committed, no?

Comment (mtosi) [reply Configuration/Generator/python/#OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py#:4]: doh :( sorry, I did not realise it, I'm going to clean

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py (hunk 1 above)
Comment (mmusich) [reply Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py]: >  By the way, why TuneCUEP8M1? Not CP5?

was this addressed? @mtosi

Comment (srimanob) [reply Configuration/Generator/python/OmegaMinus_13p6TeV_SoftQCDInel_TuneCUEP8M1_cfi.py]: Yes. Please move to Run3 tuning, Thanks.

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@mtosi I am now very confused by the state of this PR:
   * why there are now 2 configuration files for omega - ?
   * why most of the other configurations still use  `from Configuration.Generator.MCTunes2017.PythiaCP5Settings_cfi import *` ?
   * why weren't all of them changed to use `from Configuration.Generator.MCTunesRun3ECM13p6TeV.PythiaCP5Settings_cfi import *` as suggested at https://github.com/cms-sw/cmssw/pull/39178#discussion_r959509544  ?

**Comment by mtosi** [comment]
@cmsbuild, please test

File: Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import FWCore.ParameterSet.Config as cms
 
 from Configuration.Generator.Pythia8CommonSettings_cfi import *
-from Configuration.Generator.MCTunes2017.PythiaCP5Settings_cfi import *
+from Configuration.Generator.MCTunesRun3ECM13p6TeV.PythiaCP5Settings_cfi import *
 
 
 generator = cms.EDFilter("Pythia8GeneratorFilter",
```
Comment (srimanob) [inline Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py:7]: Is this file the same as 
https://github.com/cms-sw/cmssw/pull/39243

If yes, I can close that PR. And please update to `Pythia8ConcurrentGeneratorFilter`.

Comment (mmusich) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py:7]: it contains explicitly CP5

Comment (srimanob) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py:7]: I think we keep calling CP5 in production also,
https://cms-pdmv.cern.ch/mcm/requests?member_of_campaign=Run3Summer22GS&page=0&shown=549756076159

Comment (mmusich) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py:7]: well, is that the right decision?

Comment (srimanob) [reply Configuration/Generator/python/XiMinus_13p6TeV_SoftQCDInel_TuneCP5_cfi.py:7]: @cms-sw/pdmv-l2 @cms-sw/generators-l2 may clarify for the dataset/fragment names.

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by srimanob** [comment]
I closes https://github.com/cms-sw/cmssw/pull/39243 as you include the update in this PR.

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
kindly pinging @cms-sw/generators-l2 for a review (and possibly sign the PR)

**Comment by menglu21** [comment]
hi @mtosi , sorry for the late reply, btw, i see you only modified one fragment of those using evtgen to use concurrent mode, why not use what you applied in [DStarToD0Pi_D0ToKsPiPi_inclusive_SoftQCD_TuneCP5_13p6TeV-pythia8-evtgen.py](https://github.com/cms-sw/cmssw/pull/39178/files#diff-2e043f17e304964ebe7b4796a78e36c9841e31007c3b131c5e57709401870dec), i.e., wrap up the external decay using "ExternalGeneratorFilter"

**Comment by menglu21** [comment]
hi @mtosi , sorry for the late reply, btw, i see you only modified one fragment of those using evtgen to use concurrent mode, why not use what you applied in [DStarToD0Pi_D0ToKsPiPi_inclusive_SoftQCD_TuneCP5_13p6TeV-pythia8-evtgen.py](https://github.com/cms-sw/cmssw/pull/39178/files#diff-2e043f17e304964ebe7b4796a78e36c9841e31007c3b131c5e57709401870dec), i.e., wrap up the external decay using "ExternalGeneratorFilter"

**Comment by Saptaparna** [comment]
please test

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 5]
```diff
@@ -600,6 +600,7 @@ def miniAOD_customizeData(process):
     process.load("Geometry.VeryForwardGeometry.geometryRPFromDB_cfi")
     process.load('L1Trigger.L1TGlobal.simGtExtFakeProd_cfi')
     task = getPatAlgosToolsTask(process)
+    task.add(process.simGtExtUnprefireable)
```
Comment (mandrenguyen) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:603]: I suspect a rebase is needed

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by menglu21** [comment]
look fine to GEN

**Comment by mmusich** [comment]
> look fine to GEN

could you please +1 it, then? 
Thanks!

**Comment by rappoccio** [comment]
+1

**Pull request merged**